api/document/js/Api.ts

```diff
@@ -18,31 +18,60 @@ export function setStatusError(e: Error) {
   setStatus(errMsg, 'editor-status-error');
 }
 
-export default class Api {
+export interface ApiNode {
+  children: ApiNode[];
+  content: ApiContent[];
+  marker?: string;
+  node_type: string;
+  type_emblem?: string;
+  title?: string;
+  identifier?: string;
+  text?: string;
+}
```

Comment:
One minor concern I have is that there are differences in the shape of the objects the API returns vs. what the PUT endpoint expects.  For example, every `ApiNode` returned by the `GET` endpoint always has an `identifier`, but because identifiers are auto-assigned by the `PUT` endpoint, it's not required when passing the document to that endpoint.  However, this results in requiring a bunch of unnecessary assertions in code that handles the `GET` endpoint, in order to appease TypeScript.

We might be able to separate this into two different interfaces, one of which is a "stronger" version of the other, perhaps via interface inheritance or some other means.

Reply:
Excellent point; I think having two interfaces might help us telegraph the intent of the objects we're creating, too. This one'll be used when you're loading data, this one when you're saving.